tasks/collect.sh

```diff
@@ -1,42 +1,91 @@
 #!/bin/bash
 
-temp_module_directory=/var/tmp/puppet_modules
-output_directory=/var/tmp/health_check_lite
-output_file=$output_directory/health_check_lite.txt
-support_script_output_file=$output_directory/support_script_output.log
+# Variables that need to be interpolated as part of the command won't show up here
+# Should still be useful
+_debug () {
+  echo "DEBUG: running $@" >>"$_tmp.debug"
+}
 
-echo "Collecting HealthCheck Lite"
+declare PT__installdir
+source "$PT__installdir/healthcheck_lite/files/common.sh"
+[[ $PATH =~ "/opt/puppetlabs/bin" ]] || export PATH="/opt/puppetlabs/bin:${PATH}"
 
-mkdir -p $output_directory
-rm -f $output_directory/*.gz
+shopt -s nullglob extglob globstar || fail "This utility requires Bash >=4.0"
+trap '_debug $BASH_COMMAND' DEBUG
 
-echo "Puppet Enterprise HealthCheck Lite: $(date)" > $output_file
-echo >> $output_file
+(( $EUID == 0 )) || fail "This utility must be run as root"
 
-grep -i -v UUID /etc/puppetlabs/license.key >> $output_file 2>/dev/null
-echo >> $output_file
+tmp_dir=/var/tmp/puppet_modules
+output_dir=/var/tmp/health_check_lite
+output_file="$output_dir/health_check_lite.txt"
+support_script_output_file="$output_dir/support_script_output.log"
+# Not currently a parameter
+ticket="${ticket:-HCL}"
```

Comment:
Let's make `ticket` and `upload` parameters, allowing automatic uploads with new versions of PE.

Reply:
Is `--upload` something we could add later once it's more widely adopted?  I'm working on it, but it's adding complexity.  There's also the issue of the report generated by this module not being included in the tarball, so we'd either have to change the behavior, lay down the files in a way that the support script picks them up, or have the user just upload the report.  Not sure if any of these are good options.

Ticket is easy, though.

Reply:
I thought we add the text file to the tarball?

Reply:
After the fact, yes, but wouldn't it upload the file generated by `puppet enterprise support` immediately after creating it?